### PR TITLE
Added FreeBSD Support, Optimizations, Puppet 3.0 Support

### PR DESCRIPTION
**Pull request by xironix**
With these changes, I optimized the rsync setup to only have to run once, added FreeBSD support, moved OS-specific tasks to their own classes, added OS detection (required by OS-specific tasks), added version support (for installation) for puppet/librarian, hiera support (for puppet 3.0). Also changed 'restart' to 'service' in order to allow more service control (start, stop, restart, etc...).

Goodness, suppose I added a fair amount.

### CODE REVIEW

**Comment by bfirsh** [comment]
Amazing! Thank you!

I will take a look at this next week... this week's a little short of time. :)

File: loom/tasks.py [hunk 1]
```diff
@@ -42,12 +41,23 @@ def ssh(*cmd):
...
+    elif osname in 'FreeBSD':
+        freebsd.upgrade()
 
+@task
+def service(name, command):
+    """
+    Manage a service in the form of service:"nginx, restart"
+    """
+    osname = detect_os()
+    if osname in 'Ubuntu':
```
Comment (bfirsh) [inline loom/tasks.py]: `osname == 'Ubuntu'`?

Comment (xironix) [inline loom/tasks.py]: Used to compare multiple items, I shall correct that along with other similar entries.

**Comment by xironix** [comment]
Anyhow, there's no rush on merging these changes. I use this locally for myself, but figured this might be of use.

**Comment by xironix** [comment]
Probably too far out of sync to cleanly merge this. I'll update my version to the current head and consider changes.

**Comment by bfirsh** [comment]
Thanks Steffen! :)

**Pull request closed without merging**